### PR TITLE
Infrastructure migrated to ACL only access for storage

### PR DESCRIPTION
**Pull request by JosiahSiegel**
* PHI storage account access keys are disabled, which also removes the ability to use SAS tokens.
* DevOps function app added to validate function app managed identity access to PHI storage
* Function app backend storage account now separate from PHI storage

### CODE REVIEW

File: docs/security/data-access.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    { permissions = "rwx", id = var.adf_uuid, type = "user", scope = "access" },
+    { permissions = "rwx", id = var.adf_uuid, type = "user", scope = "default" },
+    { permissions = "rwx", id = var.pdi_function_app_uuid, type = "user", scope = "access" },
+    { permissions = "rwx", id = var.pdi_function_app_uuid, type = "user", scope = "default" },
+    { permissions = "r-x", id = var.infrastructure_function_app_uuid, type = "user", scope = "access" }
+  ]
+}
+```
+
+**NOTE**: While setting `default` permissions updates new files, ACL propagation is required to update existing files. The easist way to do this is by right clicking the storage account in the [Azure Storage Explorer](https://azure.microsoft.com/en-us/features/storage-explorer/).
```
Comment (cwinters-usds) [inline docs/security/data-access.md:32]: thank you!

File: operations/app/terraform/modules/data_factory/pipelines.tf [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    azurerm_data_factory_dataset_binary.vdh,
+    azurerm_data_factory_linked_service_azure_blob_storage.pdi_datasa
+  ]
+
+  variables = {
+    "sink_base_path"   = "VIIS"
+    "source_base_path" = "/"
+  }
+
+  activities_json = jsonencode(
```
Comment (cwinters-usds) [inline operations/app/terraform/modules/data_factory/pipelines.tf:19]: I'm curious: do you get this JSON through an export from the data factory?

File: operations/app/terraform/modules/function_app/main.tf [hunk 3]
```diff
@@ -3,69 +3,39 @@ resource "azurerm_function_app" "pdi" {
...
-    "WEBSITE_DNS_SERVER" = "168.63.129.16"
-
-    # "DOCKER_REGISTRY_SERVER_URL"      = var.container_registry_login_server
-    # "DOCKER_REGISTRY_SERVER_USERNAME" = var.container_registry_admin_username
-    # "DOCKER_REGISTRY_SERVER_PASSWORD" = var.container_registry_admin_password
-
-    # With this variable set, clients can only see (and pull) signed images from the registry
-    # First make signing work, then enable this
-    # "DOCKER_CONTENT_TRUST" = 1
+    WEBSITE_DNS_SERVER = "168.63.129.16"
```
Comment (cwinters-usds) [inline operations/app/terraform/modules/function_app/main.tf:17]: we should remove the comment above this, as we're explicitly NOT using CDC DNS here

File: operations/app/terraform/modules/log_analytics_workspace/data.tf [hunk 4]
```diff
@@ -1,14 +1,14 @@
 locals {
   default = {
-    "function_infrastructure_app" = {
-      id    = var.function_infrastructure_app_id
-      name  = "function_infrastructure_app"
-      diags = data.azurerm_monitor_diagnostic_categories.function_infrastructure_app
+    "infrastructure_function_app" = {
```
Comment (cwinters-usds) [inline operations/app/terraform/modules/log_analytics_workspace/data.tf:3]: thank you for fixing that name :flushed:

File: operations/app/terraform/vars/dev/azure.tf [hunk 5]
```diff
@@ -1,10 +1,10 @@
 terraform {
-  required_version = ">= 1.0.5, < 1.2.0" # This version must also be changed in other environments
+  required_version = ">= 1.0.5" # This version must also be changed in other environments
 
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = ">= 2.61.0" # This version must also be changed in other environments
+      version = "= 2.99.0" # 3.0 contains breaking changes!
```
Comment (cwinters-usds) [inline operations/app/terraform/vars/dev/azure.tf:7]: hard fought comment

**Review by cwinters-usds — APPROVED** [review]
This is a big one -- and it represents a lot of work and experimentation, nice job! :tada:

File: operations/app/terraform/modules/data_factory/pipelines.tf (hunk 2 above)
Comment (JosiahSiegel) [reply operations/app/terraform/modules/data_factory/pipelines.tf:19]: yes. a `terraform refresh` updates the state to what is remote and a `terraform state show module.data_factory.etc.etc` shows the latest.

**Review by DanPaseltiner — APPROVED** [review]
This is huge! Thank you for your work on this.

**Pull request merged**